lib/httparty/response.rb

```diff
@@ -50,7 +50,15 @@ def tap
 
     def inspect
       inspect_id = ::Kernel::format "%x", (object_id * 2)
-      %(#<#{self.class}:0x#{inspect_id} parsed_response=#{parsed_response.inspect}, @response=#{response.inspect}, @headers=#{headers.inspect}>)
+
+      parsed_response_or_body =
+        begin
+          "parsed_response=#{parsed_response.inspect}"
+        rescue JSON::ParserError
```

Comment:
My gut reaction is we don't want to rescue this specific error here. We allow custom parsers and they could really raise anything. That made me think maybe we just do a full rescue printing out "parsing the response raised an error" or something, but then you lose insight into the error unless you include the error info in that text. 

I also thought maybe we just removed parsed_response here but I suspect a good portion of people use inspect to see the parsed response which rules that idea out.

Since we have a Parser class, we could allow the implementation of that to handle errors. This feels like more work than I have time for right now or you might be interested in and might be breaking. Really the Parser should return some sort of response object instead of just the parsed data. That would allow us to denote success or failure and then inspect could use success/failure to determine if parsed response could be shown. 

Maybe I'm overthinking it. Open to any other ideas.

Reply:
Thanks so much for the comments! That makes good sense.

Maybe a reasonable middle ground would be to make it non-JSON-specific by having a blanket rescue like you suggest. That would fix the real-world issue my team's run into and we won't break any use cases that rely on the parsed response being part of the inspect.

I don't know this codebase well currently, but I would guess that going the "parser response object" route risks becoming a bit of a rabbit hole, and could possibly impact any third party format handlers, if that's a thing. It does sound a bit more elegant, but I worry about it becoming a "perfect as the enemy of the good" situation.

Reply:
Just a ping since I was reminded of this PR :)